ingest-app/src/cmr/ingest/api/routes.clj

```diff
@@ -69,6 +73,24 @@
             ctx)
            {:status 200}))))
 
+(defn- add-cmr-ingest-to-context
+  "Add CMR Ingest prefix to token in the request context of the given request.
+   This function should be called on routes that will ingest into CMR.
+   The CMR Ingest prefix is used to indicates to legacy services that when
+   validating the Launchpad token, the NAMS CMR Ingest group should also be checked.
+   Ingest will only be allowed if the user is in the NAMS CMR Ingest group and
+   also has the right ACLs which is based on Earthdata Login uid."
+  [request]
+  (let [token (-> request :request-context :token)]
+    (if (> (count token) URS_TOKEN_MAX_LENGTH)
+      ;; for Launchpad token add CMR_INGEST: prefix so that legacy service
+      ;; can do extra vaidation to check if the user has signed up for
+      ;; CMR Ingest workflow.
+      (-> request
+          (update-in [:request-context :token] #(str CMR_INGEST_SEPARATOR %))
+          (update-in [:request-context] augmenter/add-user-id-and-sids-to-context))
```

Comment:
Why did we need to explicitly call this? Isn't it already added in a ring handler?

Reply:
When the ring handler is constructed, the request context does not have the "CMR_INGEST:" prefix added to the token. Since we added the prefix, we need to update the handler to use the correct token.